### ISSUE TITLE
Need to add an API warning when 2 ContainerPorts overlap hostPorts

### ISSUE THREAD

**Issue by thockin** [kubernetes/kubernetes#118417]
It's apprently allowed to say:

```
ports:
- containerPort: 123
  hostPort: 987
- containerPort: 456
  hostPort: 987
  hostIP: 1.2.3.4
```

We should issue a warning.  The first hostPort binds to all IPs, the second binds the same hostPort to a specific IP, and we do not define the semantic.

**Comment by sftim** [comment]
I wonder if for this case we might want to report that in an Event and / or via the `.status` of the Pod once running.

**Comment by thockin** [comment]
Fixed in #113245
